types/index.d.ts

```diff
@@ -1774,8 +1774,8 @@ declare module 'mongoose' {
     polygon(path: string, ...coordinatePairs: number[][]): this;
 
     /** Specifies paths which should be populated with other documents. */
-    populate<Paths = {}>(path: string | string[], select?: string | any, model?: string | Model<any, THelpers>, match?: any): QueryWithHelpers<UnpackedIntersection<ResultType, Paths>, DocType, THelpers, RawDocType>;
-    populate<Paths = {}>(options: PopulateOptions | (PopulateOptions | string)[]): QueryWithHelpers<UnpackedIntersection<ResultType, Paths>, DocType, THelpers, RawDocType>;
+    populate<Paths = {}, TRawDocType = UnpackedIntersection<ResultType, Paths>>(path: string | string[], select?: string | any, model?: string | Model<any, THelpers>, match?: any): QueryWithHelpers<TRawDocType, DocType, THelpers, TRawDocType>;
```

Comment:
I thought I had submitted this review 2 days ago when I got this issue as well 🤦‍♂️. It's my first review and I didn't know I had to submit the review after starting it again.

Am I wrong to think it's better to use the `RawDocType` in the UnpackedIntersection for the 4th argument? Because using the `ResultType`, you get the properties of a Mongoose Document. @vkarpov15

Reply:
Thanks for your comment, You are correct, and sorry for this mistake.

Reply:
It's all good man. Thanks for the PR!!